### ISSUE TITLE
Schedule.exponential applies the exponential factor on first delay

### ISSUE THREAD

**Issue by gaelrenoux** [zio/zio#2956]
I'm using 1.0.0-RC17.

When retrying a ZIO using Schedule.exponential, the first wait is not the duration given in the call to exponential, but the duration given multiplied by the factor. Is it the intended behavior ? If so, it's very counter-intuitive.

The following waits for 5 seconds before the first retry:
```scala
import zio._
import zio.clock._
import zio.console._
import zio.duration._

val z = for {
  time <- ZIO.accessM[Clock](_.clock.nanoTime)
  _ <- ZIO.accessM[Console](_.console.putStrLn((time / 1000 / 1000 / 1000).toString))
  _ <- ZIO.fail(new Exception("Failed again")).as(0)
} yield ()
z.retry(Schedule.exponential(1.second, 5))
```

**Comment by adamgfraser** [comment]
Will look at this along with #2957.

**Comment by gaelrenoux** [comment]
For this one, the fix seems simple enough (remove the `+1` in Schedule.exponential), but I didn't want to create a PR before someone would confirm it's not intended behavior.

**Comment by adamgfraser** [comment]
Resolved by #2959.
